cdap-standalone/src/test/java/co/cask/cdap/StandaloneTester.java

```diff
@@ -61,6 +62,16 @@ protected void before() throws Throwable {
     cConf.setBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, true);
     cConf.setBoolean(StandaloneMain.DISABLE_UI, true);
 
+    // Setup test case specific configurations.
+    // The system properties are usually setup by TestConfiguration class using @ClassRule
+    for (String key : System.getProperties().stringPropertyNames()) {
```

Comment:
this is related. The test cases specify configuration parameters and values, and this is where/how they get detected and put in the CConfiguration